framework/set/resources/providers/harvester/instances.go

```diff
@@ -0,0 +1,167 @@
+func getPublicSSHKey(privateKeyPath string) string {
+	sshBytes, err := os.ReadFile(privateKeyPath)
+	if err != nil {
+		log.Fatalf("Failed to read private key file: %v", err)
+	}
+
+	signer, err := ssh.ParsePrivateKey(sshBytes)
+	if err != nil {
+		log.Fatalf("Failed to parse OpenSSH private key: %v", err)
+		return ""
+	}
+
+	publicKey := signer.PublicKey()
+
+	// Convert the public key to authorized_keys format
+	pubKeyString := string(ssh.MarshalAuthorizedKey(publicKey))
+
+	return pubKeyString
+}
+
+// CreateHarvesterInstances is a function that will set the Harvester instances configurations in the main.tf file.
+func CreateHarvesterInstances(rootBody *hclwrite.Body, terraformConfig *config.TerraformConfig, terratestConfig *config.TerratestConfig,
+	hostnamePrefix string) {
+
+	configBlockSSHKey := rootBody.AppendNewBlock(defaults.Resource, []string{defaults.HarvesterSSHKey, hostnamePrefix + "ssh_key"})
+	configBlockSSHKeyBody := configBlockSSHKey.Body()
+
+	configBlockSSHKeyBody.SetAttributeValue(defaults.Name, cty.StringVal(namegenerator.AppendRandomString("tfpsshkey")))
+	configBlockSSHKeyBody.SetAttributeValue(defaults.Namespace, cty.StringVal(terraformConfig.HarvesterConfig.VMNamespace))
+
+	publicKey := getPublicSSHKey(terraformConfig.PrivateKeyPath)
+	configBlockSSHKeyBody.SetAttributeValue(defaults.PublicKey, cty.StringVal(publicKey))
+
+	configBlockSecret := rootBody.AppendNewBlock(defaults.Resource, []string{defaults.KubernetesSecret, hostnamePrefix + "secret"})
+	configBlockSecretBody := configBlockSecret.Body()
+
+	secretBlockMeta := configBlockSecretBody.AppendNewBlock("metadata", []string{})
+	secretBlockMetaBody := secretBlockMeta.Body()
+
+	secretName := namegenerator.AppendRandomString("tfpsecret")
+
+	secretBlockMetaBody.SetAttributeValue(defaults.Name, cty.StringVal(secretName))
+	secretBlockMetaBody.SetAttributeValue(defaults.Namespace, cty.StringVal(terraformConfig.HarvesterConfig.VMNamespace))
+
+	secretBlockMetaBody.SetAttributeValue(defaults.Labels, cty.ObjectVal(map[string]cty.Value{
+		"sensitive": cty.StringVal("false"),
+	}))
+
+	hclLocalValue := hclwrite.Tokens{
+		{Type: hclsyntax.TokenIdent, Bytes: []byte("{\"userdata\" = local." + defaults.CloudInit + "}")},
+	}
+
+	configBlockSecretBody.SetAttributeRaw("data", hclLocalValue)
+
+	configBlock := rootBody.AppendNewBlock(defaults.Resource, []string{defaults.HarvesterVirtualMachine, hostnamePrefix})
+	configBlockBody := configBlock.Body()
+
+	// not sure what this does yet, but it was for aws
+	if strings.Contains(terraformConfig.Module, "custom") {
+		configBlockBody.SetAttributeValue(defaults.Count, cty.NumberIntVal(terratestConfig.NodeCount))
+	}
+
+	configBlockBody.AppendNewline()
+
+	formattedList := hclwrite.Tokens{
+		{Type: hclsyntax.TokenIdent, Bytes: []byte(`[` + fmt.Sprintf("kubernetes_secret.%s", hostnamePrefix+"secret") + `]`)},
+	}
+	configBlockBody.SetAttributeRaw("depends_on", formattedList)
```

Comment:
We have a defaults entry that you can use for the `depends_on` here. Also, can you have a newline separator for code readability?